tests/foundry/foundry.k.check.expected

```diff
@@ -3016,64 +3015,157 @@ module LOOPSTEST-BIN-RUNTIME
+    
+    syntax MethodDisambiguateTestContract ::= "MethodDisambiguateTest" [symbol(), klabel(contract_MethodDisambiguateTest)]
+    
+      
+    
+    syntax Field ::= MethodDisambiguateTestField
+    
+    syntax MethodDisambiguateTestField ::= "IS_TEST" [symbol(), klabel(field_MethodDisambiguateTest_IS_TEST)]
+    
+    syntax MethodDisambiguateTestField ::= "_failed" [symbol(), klabel(field_MethodDisambiguateTest__failed)]
+    
+    syntax MethodDisambiguateTestField ::= "IS_SCRIPT" [symbol(), klabel(field_MethodDisambiguateTest_IS_SCRIPT)]
+    
+    syntax MethodDisambiguateTestField ::= "stdstore" [symbol(), klabel(field_MethodDisambiguateTest_stdstore)]
+    
+    rule  ( #loc ( MethodDisambiguateTest . IS_TEST ) => 0 )
+      
+    
+    rule  ( #loc ( MethodDisambiguateTest . _failed ) => 0 )
+      
+    
+    rule  ( #loc ( MethodDisambiguateTest . IS_SCRIPT ) => 0 )
+      
+    
+    rule  ( #loc ( MethodDisambiguateTest . stdstore ) => 1 )
+      
+    
+    syntax Bytes ::= MethodDisambiguateTestContract "." MethodDisambiguateTestMethod [function(), symbol(), klabel(method_MethodDisambiguateTest)]
+    
+    syntax MethodDisambiguateTestMethod ::= "IS_SCRIPT" "(" ")" [symbol(), klabel(method_MethodDisambiguateTest_IS_SCRIPT_)]
+    
+    syntax MethodDisambiguateTestMethod ::= "IS_TEST" "(" ")" [symbol(), klabel(method_MethodDisambiguateTest_IS_TEST_)]
     
-    rule  ( selector ( "testIsPrimeOpt" ) => 404719852 )
+    syntax MethodDisambiguateTestMethod ::= "failed" "(" ")" [symbol(), klabel(method_MethodDisambiguateTest_failed_)]
+    
+    syntax MethodDisambiguateTestMethod ::= "getNumber" "(" Int ")" [symbol(), klabel(method_MethodDisambiguateTest_getNumber_Int)]
+    
+    syntax MethodDisambiguateTestMethod ::= "getNumber" "(" Int ")" [symbol(), klabel(method_MethodDisambiguateTest_getNumber_Int)]
+    
+    syntax MethodDisambiguateTestMethod ::= "test_method_call" "(" ")" [symbol(), klabel(method_MethodDisambiguateTest_test_method_call_)]
+    
+    syntax MethodDisambiguateTestMethod ::= "vm" "(" ")" [symbol(), klabel(method_MethodDisambiguateTest_vm_)]
+    
+    rule  ( MethodDisambiguateTest . IS_SCRIPT ( ) => #abiCallData ( "IS_SCRIPT" , .TypedArgs ) )
+      
+    
+    rule  ( MethodDisambiguateTest . IS_TEST ( ) => #abiCallData ( "IS_TEST" , .TypedArgs ) )
       
     
-    rule  ( selector ( "testMax" ) => 1575104263 )
+    rule  ( MethodDisambiguateTest . failed ( ) => #abiCallData ( "failed" , .TypedArgs ) )
       
     
-    rule  ( selector ( "testMaxBroken" ) => 1372438930 )
+    rule  ( MethodDisambiguateTest . getNumber ( V0_x ) => #abiCallData ( "getNumber" , #uint256 ( V0_x ) , .TypedArgs ) )
+       ensures #rangeUInt ( 256 , V0_x )
       
     
-    rule  ( selector ( "testNthPrime" ) => 2702762242 )
+    rule  ( MethodDisambiguateTest . getNumber ( V0_x ) => #abiCallData ( "getNumber" , #uint32 ( V0_x ) , .TypedArgs ) )
+       ensures #rangeUInt ( 32 , V0_x )
```

Comment:
Hmmmm, how will K be able to tell which `getNumber` should apply here? It seems not really possible?

Reply:
Note that getting this wrong is pretty dangerous, if we leave it as is, K may take the wrong rule, and put the wrong restrictions on the input types.

Reply:
The rules should be distinguishable now.